### PR TITLE
Change Carbon::setTestNow() to be able to parse a string.

### PR DESCRIPTION
**Pull request by c-jm**
See #817

### CODE REVIEW

**Comment by briannesbitt** [comment]
Can you add the optional string param to the phpdoc?

File: src/Carbon/Carbon.php [hunk 1]
```diff
@@ -1030,8 +1030,13 @@ public static function setWeekendDays($days)
      *
      * @param \Carbon\Carbon|null $testNow
      */
-    public static function setTestNow(Carbon $testNow = null)
+    public static function setTestNow($testNow = null)
     {
+        if (is_string($testNow)) {
```
Comment (lucasmichot) [inline src/Carbon/Carbon.php]: Why not just:
```php
$testNow = is_string($testNow) ? static::parse($testNow) : $testNow;
````

File: src/Carbon/Carbon.php [hunk 2]
```diff
@@ -1030,9 +1031,9 @@ public static function setWeekendDays($days)
      *
      * @param \Carbon\Carbon|null $testNow
```
Comment (lucasmichot) [inline src/Carbon/Carbon.php]: `@param \Carbon\Carbon|string|null $testNow`

Comment (lucasmichot) [inline src/Carbon/Carbon.php]: Ping @c-jm

**Comment by lucasmichot** [comment]
Merged via https://github.com/briannesbitt/Carbon/issues/847

**Pull request closed without merging**